File: my-folder/0636-exclusive-time-of-functions/solution.py
class Solution:
    def exclusiveTime(self, n: int, logs: List[str]) -> List[int]:
        ans = [0]*n
        stk = []
        prev = 0
        for log in logs:
            idx, op, time = log.split(':')
            idx, time = int(idx), int(time)
            if op == 'start':
                if stk:
                    ans[stk[-1]] += time - prev
                stk.append(idx)
                prev = time
            else:
                ans[stk.pop()] += time - prev + 1
                prev = time+1
        return ans

File: my-folder/0636-exclusive-time-of-functions/solution.java
class Solution {
    public int[] exclusiveTime(int n, List<String> logs) {
        int[] result = new int[n];
        
        Deque<Log> stack = new ArrayDeque<>();
        for (String content : logs) {
            Log log = new Log(content);
            
            if (log.isStart) {
                stack.push(log);
            } else {
                Log top = stack.pop();
                result[top.id] += log.time - top.time + 1;
                if (!stack.isEmpty()) {
                    result[stack.peek().id] -= (log.time - top.time + 1);
                }
            }
        } 
        
        return result;
    }
    
    public static class Log {
        public int id;
        public boolean isStart;
        public int time;
        
        public Log(String content) {
            String[] strs = content.split(":");
            id = Integer.valueOf(strs[0]);
            isStart = strs[1].equals("start");
            time = Integer.valueOf(strs[2]);
        }
    }
}
